kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java

```diff
@@ -320,9 +523,42 @@ private void generateMeaningForm() {
 
     @FXML
     private void clearView(ActionEvent actionEvent) {
+        displayNameTextField.setText(StringUtils.EMPTY);
+    }
+
+    @FXML
+    private void onCancel(ActionEvent actionEvent) {
+    }
+
+    private void collectFormData() {
+        patternFieldsViewModel.setPropertyValue(DISPLAY_NAME, displayNameTextField.getText());
+        //TODO collect all the data
     }
 
     @FXML
-    private void cancel(ActionEvent actionEvent) {
+    public void onDone(ActionEvent actionEvent) {
+        actionEvent.consume();
+        collectFormData();
+
+        // save calls validate
+        patternFieldsViewModel.save();
+
+        //publish close env
+        eventBus.publish(patternFieldsViewModel.getPropertyValue(PATTERN_TOPIC),
+                new PatternPropertyPanelEvent(actionEvent.getSource(), CLOSE_PANEL));
+
+
+        //publish form submission data
+        PatternField patternField = new PatternField(
+                patternFieldsViewModel.getPropertyValue(FIELD_ORDER),
+                patternFieldsViewModel.getPropertyValue(DISPLAY_NAME),
+                patternFieldsViewModel.getPropertyValue(DATA_TYPE),
+                patternFieldsViewModel.getPropertyValue(PURPOSE_ENTITY),
+                patternFieldsViewModel.getPropertyValue(MEANING_ENTITY),
+                patternFieldsViewModel.getPropertyValue(COMMENTS)
```

Comment:
Can you use `.getValue()` instead of `.getPropertyValue()`? 
After view model's `save()` is invoked the model values are what you want.

Assuming you have no error messages.

Reply:
fixed